data_diff/cloud/data_source.py

```diff
@@ -64,7 +68,25 @@ def create_ds_config(ds_config: TCloudApiDataSourceConfigSchema, data_source_nam
     )
 
 
-def _parse_ds_credentials(ds_config: TCloudApiDataSourceConfigSchema, only_basic_settings: bool = True):
+def _parse_ds_credentials(
+    ds_config: TCloudApiDataSourceConfigSchema, only_basic_settings: bool = True, dbt_parser: Optional[DbtParser] = None
+):
+    def cast_value(value: str, type_: str):
+        if type_ == "integer":
+            return int(value)
+        elif type_ == "boolean":
+            return bool(value)
+        return value
```

Comment:
Oh, it looks interesting. I know C/C++ has overriding mechanism, and it is quite convenient in contrast to Python.

I see that `from typing import override` fails for now because the corresponding [PEP-698](https://peps.python.org/pep-0698/) will be since Python 3.12 only...

Reply:
Ah, okay, maybe `from typing_extensions … ` then. Anyway, I have this working since Python 3.7 successfully.

Reply:
Thanks! I updated.